### PR TITLE
Add label stitching functionality

### PR DESCRIPTION
**Pull request by anwai98**
This PR adds a functionality to stitch labels to merge instance segmentations.

@constantinpape: Should we add tests for this?

### CODE REVIEW

File: elf/segmentation/stitching.py [hunk 1]
```diff
@@ -175,4 +185,127 @@ def _compute_overlaps(block_id):
...
+
+    # Get the tiles from the segmentation of shape: 'tile_shape'.
+    def _fetch_tiles(block_id):
+        block = blocking.getBlock(block_id)
+        bb = tuple(slice(beg, end) for beg, end in zip(block.begin, block.end))
+        block_seg = segmentation[bb]
+        block_segs.append(block_seg)
+
+    n_threads = multiprocessing.cpu_count() if n_threads is None else n_threads
+    with futures.ThreadPoolExecutor(n_threads) as tp:
```
Comment (constantinpape) [inline elf/segmentation/stitching.py:228]: This is premature optimization, you don't need to wrap it in a Threadpool as this is just accessing data, which is very fast.

**Comment by constantinpape** [comment]
> @constantinpape: Should we add tests for this?

Yeah, it would be good to add a small test. You can do something quite simple: create a segmenation, e.g. via  applying connected components to `skimage.data.binary_blobs`. Then add artificial tiling by adding id offsets to four tiles in the image. Then apply the stitching function and check that the result is the same as the initial segmentation.

File: elf/segmentation/stitching.py (hunk 1 above)
Comment (anwai98) [reply elf/segmentation/stitching.py]: Updated it to extract the data block wise. Thanks!

**Comment by anwai98** [comment]
I added a simple test as discussed, and I think I can reproduce some issues I spotted (which is good to see already). I kept the test checks limited to avoid the obvious directions where the tests would fail (eg. matching the true number of instances with the merged instances)

File: test/segmentation/test_stitching.py [hunk 2]
```diff
@@ -1,15 +1,48 @@
...
+
+        # Create tiles out of the data.
+        # Ensure offset for objects per tile to get individual ids per object per tile.
+        import nifty.tools as nt
+        blocking = nt.blocking([0] * ndim, data.shape, tile_shape)
+        n_blocks = blocking.numberOfBlocks
+
+        offset = 0
+        bb_tiles, tiles = [], []
+        for tile_id in range(n_blocks):
```
Comment (constantinpape) [inline test/segmentation/test_stitching.py:27]: It would be a bit more elegant to just modify the labels in place here.

**Review by constantinpape — COMMENTED** [review]
The tests looks good. We can discuss the issues.

File: test/segmentation/test_stitching.py (hunk 2 above)
Comment (anwai98) [reply test/segmentation/test_stitching.py:27]: Yes, true. I'll take care of it. Thanks!

**Comment by constantinpape** [comment]
The tests fail because test data from EMBL owncloud currently is not available. The other tests pass now, so I will merge it.

**Pull request merged**